### DISCUSSION TITLE
How does the value of a Matrix field translate for different sites in Craft5?

### DISCUSSION THREAD

**Discussion by scoch5** [craftcms/cms#15959] [Q&A] [answered]
I'm importing data from a complex excel file. I need to create or update Entries on the various sites that exist on my Craft installation. I have some Martix fields that are translatable by site but, unlike the other fields, I cannot translate their contents.
How do you translate the contents of the Matrix fields for the various sites? Do we need to make queries and update the translations of the different entries or is there a simpler way?

**Comment by brandonkelly** [comment, marked as answer]
> I have some Martix fields that are translatable by site

Matrix fields themselves are not translatable; instead they have **Propagation Methods**, which determine how their nested entries should get propagated across the sites that the owner element belongs to.

For example, if you have two sites, Site A and Site B, and a Section whose Propagation Method is set to “Save entries to all sites enabled for this section”, any entries within the section will exist across both sites. If you have a Matrix field within that section whose Propagation Method is set to “Save entries to all sites the owner element is saved in”, then its nested entries will also be saved to both sites.

Then you have fields that live within your nested Matrix entries, who may have **Translation Method** settings. Translation Methods determine whether field values should be managed independently for each site, or kept consistent between the sites.

If you have a Plain Text field set to “Not translatable”, any content saved to that field will be copied across all the sites that the element belongs to.

If it’s set to “Translate for each site”, on the other hand, then each site will get its own unique value for that field.

So to answer your question, the setup you _probably_ want (if I understand what you’re trying to get at) is:

- Set your section’s Propagation Method setting to “Save entries to all sites enabled for this section”.
- Set your Matrix field’s Propagation Method setting to “Save entries to all sites the owner element is saved in”.
- For whichever fields you wish to be translatable, set their Translation Method settings to “Translate for each site”.

**Reply by scoch5** [reply]
@brandonkelly is it possible to update a matrix field by inserting the translations already? If it is not possible, what is the way to do it?

**Reply by brandonkelly** [reply]
Can you elaborate? Not sure I’m understanding the question.

**Reply by scoch5** [reply]
I have a "Product" object that has a "highlights" matrix field. The matrix elements are saved on all sites where the proprietary element is present. Each hightlight has a title that can be translated for each site.
How can I, in PHP, save the Highlights field of a Product already inserting the translations of the title of each element?
If this is not possible, what is the best way to translate the title of these Highlight Entry?

I referred to this page https://craftcms.com/docs/5.x/reference/field-types/matrix.html#saving-matrix-fields but there is no mention of translations.

Thanks a lot

**Reply by brandonkelly** [reply]
There’s no way to save an entry across multiple sites at the same time. You need to fetch the entry for each site you want to edit it in, and update each of their localized values individually.
